### ISSUE TITLE
AMO: Tier-3 platform BSD equivalence to Tier-1 Linux for new front-end purposes

### ISSUE THREAD

**Issue by grahamperrin** [mozilla/addons-frontend#3513]
Follow-up to: 

- [Unable to install add-on on *BSD · Issue #5647 · mozilla/addons-server](https://github.com/mozilla/addons-server/issues/5647) 
- <https://github.com/mozilla/addons-server/pull/6157> (2017-08-11), which fixed [Various extensions inexplicably not, or no longer, installable to Firefox on FreeBSD through the GUI to addons.mozilla.org (AMO) · Issue #468 · mozilla/addons](https://github.com/mozilla/addons/issues/468) for the _legacy frontend_ component
- https://github.com/mozilla/addons-server/issues/6680 (misplaced). 

Shared understandings: 

> We don't officially support BSD, but this small change should improve the life of its users by letting them install add-ons marked as compatible with Linux instead of "All platforms".

> … I don't mind doing it now because it's relatively easy, but the code is old and fragile and I can't promise it won't break again in the future …

With Firefox 56.0.1 (64-bit) on FreeBSD-CURRENT **switched to the (mobile) new front-end**, there's no platform equivalence. I have not tested that end often enough to tell whether (a) it has not gained a fix for equivalence; or (b) a fix has been broken. 

In any case: @diox and colleagues, please, might there be a fix? With the same shared understandings, for the _new_, as we had for the _legacy_ front-end 👍 

TIA

## Tested

[Close Other Windows](https://addons.mozilla.org/addon/close-other-windows/) with reference to [Close Other Windows 0.1 with Firefox on FreeBSD · Issue #1 · mystor/close-other-tabs](https://github.com/mystor/close-other-tabs/issues/1).

Extensions in the attached screen recording: 

- [Context Plus](https://addons.mozilla.org/addon/context-plus/) 
- [MultiSelect Tabs](https://addons.mozilla.org/addon/multiselect-tabs/) 
- [Tab Center Redux](https://addons.mozilla.org/addon/tab-center-redux/) (featured by Mozilla)  
- [Tab notifier](https://addons.mozilla.org/addon/tab-notifier/) (featured by Mozilla) – [home page not available](http://unusoft.it/blog)
- [Webcompat.com Reporter](https://addons.mozilla.org/addon/webcompatcom-reporter/) 

– although (sorry) I can't tell/recall whether any of those five is truly incompatible with BSD.  

[2017-10-13 02:24:17 new front-end without platform equivalence.mp4.zip](https://github.com/mozilla/addons-server/files/1385100/2017-10-13.02.24.17.new.front-end.without.platform.equivalence.mp4.zip)

## Workarounds

Either: 

a) a platform-oriented override with legacy [User Agent Overrider](https://addons.mozilla.org/addon/user-agent-overrider/), if Firefox is less than 57; or 

b) _View classic desktop site_, for as long as that will be possible.

**Comment by tofumatt** [comment]
This issue is hard to understand because there's a LOT of info and links in your post without a lot of context 😄

Could you explain what the issue you're encountering is, what the steps to reproduce are, and what you expected? Otherwise we won't be able to help as I can't quite tell what's happening. It seems the issue is you want add-ons for Linux to be marked as compatible with any *NIX platform or at least BSD. I could imagine this might be tricky as OS X might could as *NIX.

Anyway, could you answer the questions I had above so we can understand the issue better? Cheers!

**Comment by grahamperrin** [comment]
Good points, thanks. 

I could/should have done more with the screen recording (a voice-over might have helped, but I'm a mumbler and that's not a great thing when (yes) there's log going on). 

Texts were relatively verbose only because I didn't want to omit, from the jigsaw, anything might have been a decision-making factor. Anyway :-) …

----

For the purposes of this issue: I agree, let's _not_ put BSD and Mac OS X under any shared umbrella. 

tl;dr the port of Firefox to FreeBSD uses code for Linux so it is, in most cases, reasonable and safe to: 

- treat Firefox on FreeBSD as equivalent to Firefox on Linux. 

It seems that at least some of the new front end is without that equivalence. The screen recording demonstrates one aspect: 

- for some compatible extensions that are enabled, a front-end **inability to disable**. 

This frame from the recording might be a good way of putting things in context: 

![2017-10-13 02 24 48 frame](https://user-images.githubusercontent.com/192271/31697947-6ee48a0a-b3b2-11e7-9b90-16f8c220ec6d.png)

– for a moment, ignore the red alert. There's a **no entry icon at the 'off' side of the green switch**. 

(That behaviour in the switching area is merely a symptom, not the underlying issue/problem.)

If the platform equivalence can be allowed for new front-end purposes, then:

- there'll be no platform-related red alert, and so on

– users of Firefox on FreeBSD will (re)gain a non-complicated method of installing, disabling and re-enabling compatible add-ons. 

----

I'm often loitering in `#firefox` and other channels so if there's too much or too little above, feel free to ping me in a channel, or to one side. Thanks. 

<http://www.freshports.org/www/firefox> for port details, only if you need them. The **pkg-message** section, not far from the head of the page, is a concise intro to how the port differs from Firefox on other platforms.

**Comment by kumar303** [comment]
> tl;dr the port of Firefox to FreeBSD uses code for Linux so it is, in most cases, reasonable and safe to:
> * treat Firefox on FreeBSD as equivalent to Firefox on Linux.

Thanks for the tl;dr. This was a known bug fixed in https://github.com/mozilla/addons-frontend/issues/2998#issuecomment-338251430 It will be released on Thursday.

**Comment by grahamperrin** [comment]
Re: <https://github.com/mozilla/addons-server/issues/2661#issuecomment-364654943> should this issue be reopened for the new look AMO?
